.github/workflows/CI-workflow.yml

```diff
@@ -133,6 +133,7 @@ jobs:
           if docker pull opensearchstaging/opensearch:$docker_version
           then
             echo "FROM opensearchstaging/opensearch:$docker_version" >> Dockerfile
+            echo "RUN if [ -d /usr/share/opensearch/plugins/opensearch-skills ]; then /usr/share/opensearch/bin/opensearch-plugin remove opensearch-skills; fi" >> Dockerfile
```

Comment:
Tested in my end, by creating a docker file:

```
FROM opensearchstaging/opensearch:2.12.0
RUN if [ -d /usr/share/opensearch/plugins/opensearch-skills ]; then /usr/share/opensearch/bin/opensearch-plugin remove opensearch-skills; fi
RUN if [ -d /usr/share/opensearch/plugins/opensearch-ml ]; then /usr/share/opensearch/bin/opensearch-plugin remove opensearch-ml; fi
COPY opensearch-ml-2.12.0.0-SNAPSHOT.zip /tmp/
RUN /usr/share/opensearch/bin/opensearch-plugin install --batch file:/tmp/opensearch-ml-2.12.0.0-SNAPSHOT.zip
```

And then ran the command: `docker build -t custom-opensearch .`

It was successfully built.